File: dehaze-python/app/models/schema/favorite.py
"""
收藏管理模块 Pydantic Schema

Form: 前端提交表单
VO: 返回视图对象
Query: 分页查询参数
"""

from pydantic import BaseModel, Field


class FavoriteCreateForm(BaseModel):
    targetType: str = Field(..., description="收藏对象类型(algorithm/result/dataset/image/preset)")
    targetId: int = Field(..., description="收藏对象ID")

File: dehaze-sdk-js/test/factories/favorite.ts
import { FavoriteForm, FavoriteQuery, FavoriteTargetType } from "@/api/favorite/model";
import { pageQuery } from "./common";

/** 创建添加收藏表单 */
export function createFavoriteForm(overrides: Partial<FavoriteForm> = {}): FavoriteForm {
  return {
    targetType: "algorithm",
    targetId: 1,
    ...overrides,
  };
}

/** 创建收藏查询参数 */
export function createFavoriteQuery(overrides: Partial<FavoriteQuery> = {}): FavoriteQuery {
  return pageQuery<FavoriteQuery>({
    pageNum: 1,
    pageSize: 10,
    ...overrides,
  });
}

/** 所有收藏对象类型（用于遍历测试） */
export const ALL_TARGET_TYPES: FavoriteTargetType[] = [
  "algorithm",
  "result",
  "dataset",
  "image",
  "preset",
];
